pheno_utils/questionnaires_handler.py

```diff
@@ -63,25 +128,57 @@ def transform_answers(
     code_to = transform_to.lower()
     assert code_from in ["hebrew", "english", "coding"], "transform_from must be one of 'hebrew', 'english', 'coding'"
     assert code_to in ["hebrew", "english", "coding"], "transform_to must be one of 'hebrew', 'english', 'coding'"
+
+    #if dictionary index is not tabular field name
+    if dict_df.index.name != 'tabular_field_name':
+        dict_df = dict_df.reset_index().set_index('tabular_field_name')
     
-    # the index of the dict_df is the tabular_field_name
+    # converting and formatting data coding values 
     if isinstance(dict_df.loc[tab_field_name]["data_coding"], pd.Series):
-        code_string = convert_to_string(dict_df.loc[tab_field_name]["data_coding"].iloc[0])
+        if dict_df.loc[tab_field_name]["data_coding"].nunique() == 1:
+            code_string = convert_to_string(dict_df.loc[tab_field_name]["data_coding"].iloc[0])
+        else:
+            warnings.warn("data_coding has multiple values for tabular field {tab_field_name}, please check and update dictionary")
```

Comment:
this will not work, you need to add "f" before the string

Reply:
added